stores/governance/src/main/java/com/bloxbean/cardano/yaci/store/governance/service/LocalDRepDistrService.java

```diff
@@ -0,0 +1,148 @@
+        if (epochAndTip.isEmpty()) {
+            log.error("Epoch is null. Cannot fetch protocol params");
+            return;
+        }
+
+        Integer epoch = epochAndTip.get()._2;
+        long slot = epochAndTip.get()._1.getPoint().getSlot();
+        Optional<LocalClientProvider> localClientProvider = localClientProviderManager.getLocalClientProvider();
+
+        try {
+            var localStateQueryClient = localClientProvider.map(LocalClientProvider::getLocalStateQueryClient).orElse(null);
+            if (localStateQueryClient == null) {
+                log.info("LocalStateQueryClient is not initialized. Please check if n2c-node-socket-path or n2c-host is configured properly.");
+                return;
+            }
+
+            //Try to release first before a new query to avoid stale data
+            try {
+                localStateQueryClient.release().block(Duration.ofSeconds(5));
+            } catch (Exception e) {
+                //Ignore the error
+            }
+
+            try {
+                localStateQueryClient.acquire().block(Duration.ofSeconds(5));
+            } catch (Exception e) {
+                // Ignore the error
+            }
+
+            Mono<DRepStakeDistributionQueryResult> mono = localStateQueryClient.executeQuery(new DRepStakeDistributionQuery(era, List.of()));
+            mono.doOnError(throwable ->
+                            log.error("DRep stake distribution sync error {}", throwable.getMessage()))
+                    .doFinally(
+                            signalType -> localClientProviderManager.close(localClientProvider.get()))
+                    .subscribe(dRepStakeDistributionQueryResult -> {
+                        Map<Drep, BigInteger> dRepStakeDistrMap = dRepStakeDistributionQueryResult.getDRepStakeMap();
+                        List<LocalDRepDistr> localDRepDistrList = new ArrayList<>();
+
+                        dRepStakeDistrMap.forEach((drep, amount) -> {
+                            String drepHash;
+                            if (drep.getType() == DrepType.NO_CONFIDENCE) {
+                                drepHash = "no-confidence";
+                            } else if (drep.getType() == DrepType.ABSTAIN) {
+                                drepHash = "abstain";
+                            } else {
+                                drepHash = drep.getHash();
+                            }
+
+                            localDRepDistrList.add(LocalDRepDistr.builder()
+                                    .drepHash(drepHash)
+                                    .drepType(drep.getType())
+                                    .amount(amount)
+                                    .epoch(epoch)
+                                    .slot(slot)
+                                    .build());
+                        });
+
+                        if (!localDRepDistrList.isEmpty()) {
+                            localDRepDistrStorage.saveAll(localDRepDistrList);
```

Comment:
As this method is also called from scheduler, will it ignore saving the same data if it's already available for the epoch?

Reply:
No, it will not ignore, in local_drep_distr, (drep_hash, epoch) is primary key, so if data' already available for the epoch, it will be updated

Reply:
Should we ignore update if it's already there?

Reply:
I removed scheduler, so it will not be updated